neuralnetwork/machinelearning/Train.java

```diff
@@ -2,51 +2,73 @@
 
 import neuralnetwork.dataset.Data;
 import neuralnetwork.dataset.Dataset;
+import neuralnetwork.initializemethod.HeInitialization;
 import neuralnetwork.initializemethod.InitializeMethod;
 
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.Calendar;
 
 public class Train {
+
     private NeuralNetwork neuralNetwork;
-    private Dataset trainDataset = null;
-    private Dataset testDataset = null;
-    private int batch;
-    private int epoch;
+    private Dataset trainDataset;
+    private Dataset testDataset;
+    private int batch = 4; // default option
+    private int epoch = 5; // default option
+    private boolean backup = true; // default option
     private int testBatch = 10;
     private boolean isInitialized = false;
-    private String initializeMethod;
+    private final String initializeMethodName;
+    InitializeMethod initializeMethod = TrainMethod.He.get(); // default is HeInitialization
 
-    public Train(NeuralNetwork neuralNetwork) {
+    private Train(NeuralNetwork neuralNetwork) throws Exception {
         this.neuralNetwork = neuralNetwork;
+        this.isInitialized = true;
+        this.neuralNetwork.setLayerData(initializeMethod.set(neuralNetwork));
+        this.initializeMethodName = initializeMethod.getName();
     }
 
-    public void setTestDataset(Dataset testDataset) {
+    public static Train initialize(NeuralNetwork neuralNetwork) throws Exception {
+        return new Train(neuralNetwork);
+    }
+    public Train test(Dataset testDataset) {
         this.testDataset = testDataset;
+        return this;
     }
-
-    public void setTrainDataset(Dataset trainDataset) {
+    public Train train(Dataset trainDataset) {
         this.trainDataset = trainDataset;
+        return this;
     }
 
-    public void Initialize(InitializeMethod initializeMethod) throws Exception {
-        this.isInitialized = true;
-        this.neuralNetwork.setLayerData(initializeMethod.set(neuralNetwork));
-        this.initializeMethod = initializeMethod.getName();
+    public String getInitializeMethodName() {
+        return initializeMethodName;
     }
 
-    public void run(int batch, int epoch) throws Exception {
-        this.batch = batch;
-        this.epoch = epoch;
-        this.neuralNetwork.printSetting();
-        train(false);
+    public Train useMethod(InitializeMethod initializeMethod) {
```

Comment:
Call of setLayerData, toggle isInitialized, getting name of initializeMethod shoulde be here